src/Configuration/Config.php

```diff
@@ -18,17 +20,71 @@ class Config
     /** @var PathResolver */
     private $pathResolver;
 
-    public function __construct()
+    /** @var Stopwatch */
+    private $stopwatch;
+
+    /** @var FilesystemCache */
+    private $cache;
+
+    /** @var array */
+    private $timestamps = [];
+
+    public function __construct(Stopwatch $stopwatch)
     {
+        $this->stopwatch = $stopwatch;
+        $this->cache = new FilesystemCache();
```

Comment:
please don't. cache driver must be replaceable. use cache interface as dependency or create cache decorator class

Reply:
I tried to use `CacheInterface` as DI, but couldn't get it to work. I'll update the other points you mention, and then maybe you could take a look to see how to do that?